sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -80,15 +85,20 @@ using IsReduOptForFastAtomicFetch =
                            IsReduBitOR<T, BinaryOperation>::value ||
                            IsReduBitXOR<T, BinaryOperation>::value ||
                            IsReduBitAND<T, BinaryOperation>::value)>;
+#endif
 
 template <typename T, class BinaryOperation>
 using IsReduOptForFastReduce = detail::bool_constant<
+#ifdef SYCL_REDUCTION_NO_FAST_OPTS
```

Comment:
Nit: If keeping these macros, it's a little odd that this one only encloses `false>`, but the one before it encloses the `detail::bool_constant` from both branches.  I think the one from L77 is easier to read.